File: app/EatUp_Client/app/src/main/java/com/github/eatup_client/model/Session.java
package com.github.eatup_client.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

public class Session implements Serializable {
    @SerializedName("id")
    @Expose
    private String id;
    @SerializedName("table_id")
    @Expose
    private String tableId;
    @SerializedName("in_progress")
    @Expose
    private boolean inProgress;

    // Constructor
    public Session(String id, String tableId, boolean inProgress) {
        this.id = id;
        this.tableId = tableId;
        this.inProgress = inProgress;
    }

    // Getters
    public String getId() {
        return id;
    }

    public String getTableId() {
        return tableId;
    }

    public boolean isInProgress() {
        return inProgress;
    }

    // ToString method for debugging
    @Override
    public String toString() {
        return "Session{" +
                "id='" + id + '\'' +
                ", tableId='" + tableId + '\'' +
                ", inProgress=" + inProgress +
                '}';
    }
}

File: staff_web/eatup_staff_web/src/model/api/Session.ts
class Session {
  id: string;
  table_id: string;
  in_progress: boolean;

  constructor(id: string, table_id: string, in_progress: boolean) {
    this.id = id;
    this.table_id = table_id;
    this.in_progress = in_progress;
  }

  private static fromJson(json: any): Session {
    return new Session(json.id, json.table_id, json.in_progress);
  }

  public static fromJsonArray(json: any[]): Session[] {
    return json.map((session: any) => Session.fromJson(session));
  }
}

export default Session;